lib/cloudware/models/concerns/deployment_tags.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Cloudware
+  module Models
+    module Concerns
+      module ModelTags
+        # NOTE: The TAG_TYPE must be set as a constant on the base model
+
+        def self.included(base)
+          base.extend(ClassMethods)
+        end
+
+        module ClassMethods
+          def flag
+            'TAG'
+          end
+
+          def build_from_context(context)
+            (context.results || {})
+                    .keys
+                    .map { |k| self.name_from_tag(k) }
+                    .uniq
+                    .reject { |n| n.nil? }
+                    .map do |name|
+              self.new(name: name, context: context)
+            end
+          end
+
+          def name_from_tag(tag)
+            regex = /\A.*(?=#{flag}.*\Z)/
```

Comment:
This regex needs a comment to explain what it actually matches pls.

Reply:
https://github.com/alces-software/cloudware/commit/90fc69eaa3ecfbc57b6ed4d299ed8902fb4e5b3e